frontend/packages/console-shared/src/components/dropdown/ResourceDropdown.tsx

```diff
@@ -215,26 +218,28 @@ class ResourceDropdown extends React.Component<ResourceDropdownProps, State> {
             ? actionItems[0].actionKey
             : _.get(_.keys(sortedList), 0);
       }
-      selectedItem && this.handleChange(selectedItem, sortedList);
+      selectedItem && this.handleChange(selectedItem, sortedList, resourceList);
     }
     return sortedList;
   };
 
-  private handleChange = (key, items) => {
+  private handleChange = (key, items, resourceList) => {
     const name = items[key];
+    const resource = resourceList[key];
+    this.setState({ resources: resourceList });
     const { actionItems, onChange, selectedKey } = this.props;
     const selectedActionItem = actionItems && actionItems.find((ai) => key === ai.actionKey);
     const title = selectedActionItem ? selectedActionItem.actionTitle : name;
     if (title !== this.state.title) {
       this.setState({ title });
     }
     if (key !== selectedKey) {
-      onChange && onChange(key, name, _.isEmpty(items));
```

Comment:
@andrewballantyne Modified the onChange handler in ResourceDropdown.tsx to pass the selected resource.
`.isEmpty(items)`  was introduced in early developement phase for PipelineResourceDropdownField.tsx in https://github.com/karthikjeeyar/console/commit/0f8674a0855692f0af066559502f9ee0affbde79 and is no longer needed as the PipelineResourceDropdownField's implementation got changed.

Reply:
Ah gotcha. Wasn't sure we wanted to modify this component. It's a Class component which made me think we didn't create it and thus shouldn't be cavalier modifying it... but sounds like we are good here.